dotCMS/src/main/java/com/dotcms/rest/api/v1/theme/ThemeResource.java

```diff
@@ -134,29 +144,20 @@ public final Response findThemes(@Context final HttpServletRequest request,
     @NoCache
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
     public final Response findThemeById(@Context final HttpServletRequest request,
-            @PathParam("id") final String id)
-            throws Throwable {
+            @PathParam("id") final String id) throws DotDataException, DotSecurityException {
 
         Logger.debug(this,
                 "Getting the theme by identifier: " + id);
 
         final InitDataObject initData = this.webResource.init(null, true, request, true, null);
         final User user = initData.getUser();
 
+        final Folder folder = folderAPI.find(id, user, false);
 
-        try {
-            final Map<String, Object> params = map(
-                    ThemePaginator.ID_PARAMETER, id
-            );
-
-            return this.paginationUtil.getPage(request, user, null, 0, -1, null,
-                    OrderDirection.ASC, params);
-        } catch (PaginationException e) {
-            throw e.getCause();
-        } catch (JsonProcessingRuntimeException e) {
-
-            Logger.error(this, e.getMessage(), e);
-            return ExceptionMapperUtil.createResponse(e, Status.INTERNAL_SERVER_ERROR);
+        if (folder == null) {
+            return Response.status(Status.NOT_FOUND).build();
```

Comment:
It would be good to send more information (for instance the id) also log the inputs